captcha.module

```diff
@@ -144,8 +144,7 @@ function captcha_element_process(array $element, FormStateInterface $form_state,
   // This needs to be done even if the CAPTCHA will be ommitted later:
   // other untrusted users should not get a cached page when
   // the current untrusted user can skip the current CAPTCHA.
-  global $conf;
-  $conf['cache'] = FALSE;
+  \Drupal::service('page_cache_kill_switch')->trigger();
```

Comment:
hass has a good point. Let's try and move this logic to the separate plugins (test, image and math), all of those should include this I think.